AirCastingTests/SearchAndFollowTests/SearchViewModelTests.swift

```diff
@@ -0,0 +1,56 @@
+// Created by Lunar on 03/08/2022.
+//
+import XCTest
+@testable import AirCasting
+
+class SearchViewModelTests: XCTestCase {
+    var sut: SearchViewModel!
+    
+    override func setUp() {
+        super.setUp()
+        sut = .init()
+    }
+    
+    func test_onInitialization_onlyParticulateMatterShouldBeSelected() {
+        let selected = sut.measurementTypes.filter({ $0.isSelected })
+        XCTAssertEqual(selected.count, 1)
+        XCTAssertEqual(selected.first?.name, "Particulate Matter")
+    }
+    
+    func test_whenSensorTapHappens_shouldSelectTapped() {
+        sut.sensorTypes = [.init(isSelected: false, name: "1"),
+                           .init(isSelected: true, name: "2"),
+                           .init(isSelected: false, name: "3")]
+        
+        let tappingSensorName = "1"
+        
+        sut.onSensorTap(with: tappingSensorName)
+        
+        let selected = sut.sensorTypes.filter({ $0.isSelected })
+        
+        XCTAssertEqual(selected.count, 1)
+        XCTAssertEqual(selected.first!.name, tappingSensorName)
+    }
+    
+    func test_whenAskedToDeselectAll_shouldIsSelectedBeFalse() {
```

Comment:
Okay, I have more context now. I think you are already checking the deselecting behavior in the test above (when one sensor get's selected, all of the others sensors get deselected). I think it's not necessary to test this method separately, as what matters is the final behavior.

Reply:
I see your point and don't have hard opinion here, so I will delete